hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -560,20 +579,24 @@ public HashSet<String> readRocksDBLiveFiles(String dbPathArg) {
   /**
    * Process each line of compaction log text file input and populate the DAG.
    */
-  private synchronized void processCompactionLogLine(String line) {
+  synchronized void processCompactionLogLine(String line) {
 
     LOG.debug("Processing line: {}", line);
 
     if (line.startsWith("#")) {
       // Skip comments
       LOG.debug("Comment line, skipped");
     } else if (line.startsWith(COMPACTION_LOG_SEQNUM_LINE_PREFIX)) {
-      // Read sequence number
-      LOG.debug("Reading sequence number as snapshot generation");
-      final String seqNumStr =
+      // Read sequence number, and snapshot ID
+      LOG.debug("Reading sequence number as snapshot generation, "
+          + "and snapshot ID");
+      final String trimmedStr =
           line.substring(COMPACTION_LOG_SEQNUM_LINE_PREFIX.length()).trim();
+      final Scanner input = new Scanner(trimmedStr);
       // This would the snapshot generation for the nodes to come
-      reconstructionSnapshotGeneration = Long.parseLong(seqNumStr);
+      reconstructionSnapshotGeneration = input.nextLong();
+      // This is the snapshotID assigned to every single CompactionNode to come
+      reconstructionLastSnapshotID = input.nextLine().trim();
     } else if (line.startsWith(COMPACTION_LOG_ENTRY_LINE_PREFIX)) {
       // Read compaction log entry
```

Comment:
At line # 604, `COMPACTION_LOG_ENTRY_LINE_PREFIX` should be used instead of `COMPACTION_LOG_SEQNUM_LINE_PREFIX`. Otherwise it can cause issue if we change `COMPACTION_LOG_SEQNUM_LINE_PREFIX` or `COMPACTION_LOG_ENTRY_LINE_PREFIX` to something with different length.

suggestion: 
```
line = line.substring(COMPACTION_LOG_ENTRY_LINE_PREFIX.length());
```

Reply:
Ah! Good catch. :D